server/datastore/mysql/apple_mdm.go

```diff
@@ -1461,3 +1461,32 @@ WHERE
 
 	return &res, nil
 }
+
+func (ds *Datastore) UpsertMDMAppleConfigProfile(ctx context.Context, cp fleet.MDMAppleConfigProfile) (*fleet.MDMAppleConfigProfile, error) {
+	stmt := `
+INSERT INTO
+    mdm_apple_configuration_profiles (team_id, identifier, name, mobileconfig)
+VALUES (?, ?, ?, ?) ON DUPLICATE KEY UPDATE
+  updated_at = CURRENT_TIMESTAMP,
+  mobileconfig = VALUES(mobileconfig)
+`
+	var teamID uint
+	if cp.TeamID != nil {
+		teamID = *cp.TeamID
+	}
+
+	res, err := ds.writer.ExecContext(ctx, stmt, teamID, cp.Identifier, cp.Name, cp.Mobileconfig)
+	if err != nil {
+		return nil, ctxerr.Wrapf(ctx, err, "upsert mdm config profile team=%d identifier=%s name=%s", teamID, cp.Identifier, cp.Name)
+	}
+
+	id, _ := res.LastInsertId()
```

Comment:
thank you! it won't, double thank you for this comment:

https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/mysql.go#L1127-L1149


I'm thinking I'm going to change the signature of the function to just return an error, since we're not using the config profile in the caller anyways.

Reply:
:+1: makes sense to keep it simple if we don't need it at the moment anyway.